### ISSUE TITLE
[Backend] websocket problem

### ISSUE THREAD

**Issue by knoxdj** [thm-projects/arsnova-mobile#86]
when i'm going to create session in arsnova mobile the debbuger shows socket.io.slim.min.js:7 WebSocket connection to 'ws://%22http/socket.io/?EIO=3&transport=websocket' failed: Error in connection establishment: net::ERR_NAME_NOT_RESOLVED-...and i can't save my session...what i'm doing wrong?? 
thanks for your help...

**Comment by tekay** [comment]
Hello,
do you use your custom installation? 
It seems that the path for the WebSocket connection is broken. It should be something like `ws://<domain.under.which.arsnova.runs>/socket.io [...]`.

**Comment by knoxdj** [comment]
Thanks for the answer, but I need to know what file I have to modify, arsnova.conf? Or where do I have to configure this ws ? Thanks again..

**Comment by tekay** [comment]
I'm sorry, but I still don't know what you are trying to do. Are you running arsnova locally in a development enviroment or are you using the built versions? I need at least some information about your setup/environment.

**Comment by knoxdj** [comment]
I apologize for not having explained well
I'm using arsnova.mobile and arsnova.backend in development testing in my
virtualbox local machine with ubuntu 16.04. The attachment is the
configuration for arsnova backend. Thanks again

**Comment by knoxdj** [comment]
this is my arsnova.config:
################################################################################
# General server settings
################################################################################
# The URL under which the ARSnova server is reachable. Use
# http://localhost:8080 for development.https://example.com
root-url=http://localhost:8080

# The path where the ARSnova API is accessible by clients. By default, this path
# is set to the context path of the backend. If you are running the backend
# behind a proxy server, you might need to set this path manually.
api.path=/api


# The context paths where the ARSnova modules have been deployed
customization.path=/customization
mobile.path=/mobile
presenter.path=/presenter

# WebSockets server
socketio.bind-address=localhost
socketio.port=8090
# SSL/TLS configuration
# To enable SSL for Socket.IO you have to import your key and certificate files into a Java Key Store. If you tunnel
# WebSocket connections through a proxy server, you can skip this configuration.
# See https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html
#
#socketio.ssl.jks-file=/etc/arsnova/arsnova.jks
#socketio.ssl.jks-password=arsnova
#socketio.proxy-path=/socket.io

# Admin accounts
# Usernames of the accounts which are allowed to manage global messages of the
# day. Multiple accounts are separated by commas.
security.admin-accounts=


################################################################################
# Database
################################################################################
couchdb.host=localhost
couchdb.port=5984
couchdb.name=arsnova
couchdb.username=admin
couchdb.password=1234

################################################################################
# E-Mail
################################################################################
mail.sender.name=ARSnova
mail.sender.address=
mail.host=


################################################################################
# Authentication
################################################################################
# After the specified number of login tries the client IP will be banned for
# several minutes
security.authentication.login-try-limit=50

# Configuration parameters for authentication services:
# enabled: enable or disable the service
# allowed-roles: enable/disable service for a specific role (valid roles: speaker, student)
#   (security note: this is currently only handled by the frontend!)
# title: the title which is displayed by frontends
# login-dialog-path: URL of a login dialog page
# image: an image which is used for frontend buttons

# Guest authentication
#
security.guest.enabled=true
security.guest.allowed-roles=speaker,student
security.guest.order=0

# Setup combined login if you want to use a single, customized login page
# which is used for multiple authentication services.
#
security.custom-login.enabled=false
security.custom-login.allowed-roles=speaker,student
security.custom-login.title=University
security.custom-login.login-dialog-path=
security.custom-login.image=
security.custom-login.order=0

# Internal authentication
#
# Specific parameters:
# activation-path: URL of the account activation page
# allowed-email-domains: Allows you to restrict registration to specific
#     domains. You can use wildcards (*), e. g. *.*.example.com. Multiple
#     entries are separated by commas.
# registration-mail.subject: Subject used for registration e-mail
# registration-mail.body: Text body used for registration e-mail. {0} will be
#     replaced by the value of activation-path.
#
security.user-db.enabled=true
security.user-db.allowed-roles=speaker,student
security.user-db.title=ARSnova
security.user-db.login-dialog-path=account.html
security.user-db.activation-path=account.html
security.user-db.reset-password-path=account.html
security.user-db.image=
security.user-db.order=0
security.user-db.allowed-email-domains=*
security.user-db.registration-mail.subject=ARSnova Registration
security.user-db.registration-mail.body=Welcome to ARSnova!\n\nPlease confirm \
    your registration by visiting the following web address:\n{0}\n\n\
    Afterwards, you can log into ARSnova with your e-mail address and password.
security.user-db.reset-password-mail.subject=ARSnova Password Reset
security.user-db.reset-password-mail.body=You requested to reset your \
    password.\n\nPlease follow the link below to set a new password:\n{0}

# LDAP authentication
#
# Specific parameters:
# url: LDAP server URL
# user-dn-pattern: Pattern used to check user credentials against the LDAP
#     server. {0} will be replaced with the user ID by ARSnova.
#
security.ldap.enabled=false
security.ldap.allowed-roles=speaker,student
security.ldap.title=LDAP
security.ldap.login-dialog-path=login-ldap.html
security.ldap.image=
security.ldap.order=0
security.ldap.url=ldaps://example.com:636/dc=example,dc=com
security.ldap.user-dn-pattern=uid={0},ou=arsnova
# Set the following properties if you want to use LDAP search instead of binding
# with a DN pattern
#security.ldap.user-search-filter=(uid={0})
#security.ldap.user-search-base="ou=people"
# Configure the LDAP manager user if anonymous binding is not allowed
#security.ldap.manager-user-dn=cn=arsnova-manager,dc=example,dc=com
#security.ldap.manager-password=arsnova

# CAS authentication
#
security.cas.enabled=false
security.cas.allowed-roles=speaker,student
security.cas.title=CAS
security.cas.image=
security.cas.order=0
security.cas-server-url=https://example.com/cas

# OAuth authentication with third party services
# Specific parameters:
# key: OAuth key/id provided by a third party auth service
# secret: OAuth secret provided by a third party auth service

# Facebook
#
security.facebook.enabled=false
security.facebook.allowed-roles=speaker,student
security.facebook.order=0
security.facebook.key=
security.facebook.secret=

# Twitter
#
security.twitter.enabled=false
security.twitter.allowed-roles=speaker,student
security.twitter.order=0
security.twitter.key=
security.twitter.secret=

# Google
#
security.google.enabled=false
security.google.allowed-roles=speaker,student
security.google.order=0
security.google.key=
security.google.secret=


################################################################################
# Cross-Origin Resource Sharing
################################################################################
# CORS grants full API access to client-side (browser) applications from other
# domains. Multiple entries are separated by commas. Untrusted and vulnerable
# applications running on these domains pose a security risk to ARSnova users.
#security.cors.origins=https://


################################################################################
# ARSnova Connector (for LMS)
################################################################################
connector.enable=false
connector.uri=http://localhost:8080/connector-service
connector.username=test
connector.password=test


################################################################################
# Features
################################################################################
# Enable MathJax to allow the use of Math formulas written in TeX syntax in
# text fields.
features.mathjax.enabled=true
features.mathjax.src=//cdn.mathjax.org/mathjax/2.4-latest/MathJax.js

# The following features are considered experimental because they have not been
# tested in a production environment over a longer time frame and/or their
# behavior will change in future releases.
#
features.freetext-imageanswer.enabled=true
features.question-format.grid-square.enabled=true

# Without enabled session-import-export feature no sessions can be added to the
# public pool
#
features.session-import-export.enabled=true
features.public-pool.enabled=true
features.export-to-click.enabled=true

################################################################################
# Customization
################################################################################
# Minutes, after which the feedback is deleted
feedback.cleanup=10

# Number of "you've lost me"/"abgehängt" votes, starting to show warning sign
feedback.warning=15

# Maximal filesize in bytes
upload.filesize_b=4194304

# Maximal thumbnail size in pixels
imageupload.thumbnail.width=400
imageupload.thumbnail.height=400

# Maximal number of answer options allowed for a skill question
question.answer-option-limit=8

# Enable Markdown and MathJax parsing in answer options. Formatting in answer
# options should be used cautiously since it could lead to display errors.
# Answer options will still not be parsed in diagrams. This setting has no
# effect if neither MathJax nor Markdown are enabled.
question.parse-answer-option-formatting=false

# Demo session id to show above session login button
# You can freely use the demo session of https://arsnova.eu/mobile which can be
# imported from the repro:
# https://github.com/thm-projects/arsnova-mobile/tree/master/demo-sessions
session.demo-id=

# Delete guest sessions automatically after X days of owner inactivity.
#session.guest-session.cleanup-days=180

# Label underneath ARSnova logo
ui.slogan=Audience Response System

# Splashscreen configuration
# Best results are achieved with logos that have the same width and height.
#
ui.splashscreen.logo-path=resources/images/arsnova-logo.png
ui.splashscreen.slogan=Audience Response System
ui.splashscreen.min-delay=3000

# Splash screen colors
# These settings override the default colors of the frontend theme.
#
#ui.splashscreen.slogan-color=
#ui.splashscreen.background-color=
#ui.splashscreen.loading-ind-color=

# Links which are displayed in the frontend applications
#
links.blog.url=https://arsnova.eu/blog/
links.documentation.url=https://arsnova.eu/manual/
links.presenter-documentation.url=
links.overlay.url=https://arsnova.eu/overlay/
links.organization.url=
links.imprint.url=
links.privacy-policy.url=

# Configuration for the public pool
#
pp.subjects = Allgemeines,Anglistik,Architektur,Bauingenieurwesen,Betriebswirtschaftslehre,Biologie,Chemie,Elektrotechnik,Geographie,Geowissenschaften,Germanistik,Geschichtswissenschaft,Informatik,Kunst,Kunstpädagogik,Maschinenbau,Mathematik,Medienwissenschaften,Medizin,Musik,Pharmazie,Physik,Psychologie,Rechtswissenschaften,Stadtplanung,Theaterwissenschaften,Theologie,Vermessungswesen,Volkswirtschaftslehre
pp.licenses = CC by - Attribution,\
    CC nc - Non-Commercial,\
    CC nd - No Derivatives,\
    CC sa - Share Alike,\
    CCO - Public Domain,\
    CC by-nd -  Attribution-No Derivatives,\
    CC by-c - Attribution-Non-Commercial,\
    CC by-nc-sa - Attribution-Non-Derivatives-Share Alike,\
    CC by-nc-nd Attribution-Non-Commercial- No Derivatives,\
    GNU GPL - GNU General Public License
pp.logofilesize_b = 102400
# Do not reorder levels. Append new levels to the end of the list. Otherwise,
# references in existing public sessions could be mixed up.
#
pp.session-levels.de = Allgemeinbildung,\
    Abitur,\
    Bachelor,\
    Master,\
    Wer wird Millionär,\
    Sonstiges
pp.session-levels.en = General Education,\
    Highschool,\
    Bachelor,\
    Master,\
    Who Wants to Be a Millionaire,\
    Miscellaneous


################################################################################
# Tracking
################################################################################
# It is possible to use an external tracking software with ARSnova. Currently
# Piwik is the only supported tracking provider.
#
tracking.provider=piwik
tracking.tracker-url=
tracking.site-id=

**Comment by knoxdj** [comment]
yes i did and look the response
![captura de pantalla de 2018-08-05 11-23-02](https://user-images.githubusercontent.com/41529255/43687839-5fe138e8-98a2-11e8-979c-bbc10aec9027.png)

**Comment by knoxdj** [comment]
i copied /home/jim/Documentos/Proyectos/arsnova-backend-master/src/main/resources **arsnova.properties.example** to etc/arsnova and i renamed arsnova.properties.example to arsnova.properties and i changed root-url=https://localhost:8080 and couchdb.username=admin
couchdb.password=1234

**Comment by knoxdj** [comment]
it fails exactly here:
![captura de pantalla de 2018-08-05 11-40-27](https://user-images.githubusercontent.com/41529255/43687933-962c0d7c-98a4-11e8-912b-7d494fca1d2c.png)

**Comment by knoxdj** [comment]
Sorry again for solution to  this problem Apache2 is necessarily installed? Really I'm worried about this...thanks for all the help that you can

**Comment by commana** [comment]
Hi @knoxdj, are you using the master branches on both backend and mobile? And could you check the Ajax request when you hit "Create Session": what response body do you see?

**Comment by knoxdj** [comment]
hi @commana  thanks for the feedback. I using the master branches on both like you say and the error is the same in "createSession".
![captura de pantalla de 2018-08-07 11-10-43](https://user-images.githubusercontent.com/41529255/43788756-d5825900-9a33-11e8-8463-6d41b4c998b4.png)
![captura de pantalla de 2018-08-07 11-01-16](https://user-images.githubusercontent.com/41529255/43788768-de79e4f6-9a33-11e8-8796-1b193b6d458c.png)

**Comment by commana** [comment]
Thanks for your response, @knoxdj. Could you check the HTTP requests in the Network tab of your Browser's DevTools -- what responses do you get, especially for the request that is responsible for creating the session?

**Comment by commana** [comment]
Thanks! Session creation fails with a 403 Forbidden status code. Could you show us the full response body you get for that request?

**Comment by knoxdj** [comment]
@commana  Again thans a lot. This is a text from "Response" in Devops
{"errorType":"AccessDeniedException","errorMessage":"Access is denied"}
And this is text from the tab "Preview"
{errorType: "AccessDeniedException", errorMessage: "Access is denied"}
errorMessage
:
"Access is denied"
errorType
:
"AccessDeniedException"
And the tab "Headers"
Request URL: http://localhost:8081/api/session/?_dc=1533686124753
Request Method: POST
Status Code: 403 Forbidden
Remote Address: 127.0.0.1:8081
Referrer Policy: no-referrer-when-downgrade
HTTP/1.1 403 Forbidden
connection: close
date: Tue, 07 Aug 2018 23:55:24 GMT
content-type: application/vnd.de.thm.arsnova.v3+json;charset=UTF-8
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
cache-control: no-cache, no-store, max-age=0, must-revalidate
pragma: no-cache
expires: 0
x-frame-options: DENY
server: Jetty(9.4.11.v20180605)
Transfer-Encoding: chunked
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: es-419,es;q=0.9
Connection: keep-alive
Content-Length: 279
Content-Type: application/json
Cookie: JSESSIONID=node01d4q9cswle3hc2f38z4n1szdi0.node0
Host: localhost:8081
Origin: http://localhost:8081
Referer: http://localhost:8081/mobile/
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.84 Safari/537.36
X-Requested-With: XMLHttpRequest
_dc: 1533686124753
{name: "probe", shortName: "probe", courseId: null, courseType: null, creationTime: 1533686124749,…}
courseId
:
null
courseType
:
null
creationTime
:
1533686124749
name
:
"probe"
ppAuthorMail
:
null
ppAuthorName
:
null
ppDescription
:
null
ppFaculty
:
null
ppLevel
:
null
ppLicense
:
null
ppLogo
:
null
ppSubject
:
null
ppUniversity
:
null
sessionType
:
null
shortName
:
"probe"

A last question: do i need a special permission in a three of the arsnova mobile? 
Really really THANKS A LOT...

**Comment by commana** [comment]
Thanks for providing the response body, @knoxdj! I believe that this "access denied" error is the real problem. The websocket error looks odd, but to me, the real problem is with the regular HTTP requests.

In fact, this is currently a known issue with the master branch of the arsnova-backend that we are working on. We plan to resolve this in the coming weeks.

If you could briefly tell us what you are trying to do with ARSnova, e.g. just installing and running it unmodified or preparing to develop new features etc., we could try to find some workarounds for you until the master branch is fully functional again.

**Comment by knoxdj** [comment]
@commana  Oh God...Okay..The project is from a friend and he wants the json Data produced in a couch Db to manage that data and upload to http://rageproject.eu/ (that project i don't know) to analitics, and others. Really i need the data produced when the app is online. I tested arsnova.click but i don't know how to access the data with mongodb...that's the target of all this...it's a shame that i couldn't concrete this project.. Would you guide me from arsnova.click? Thanks again from all your help.
